custom_components/xiaomi_cloud_map_extractor/common/backoff.py

```diff
@@ -0,0 +1,14 @@
+import random
+
+
+class Backoff:
+    """Exponential backoff with jitter."""
+
+    def __init__(self, min_sleep, max_sleep):
+        self.min_sleep = min_sleep
+        self.sleep = min_sleep
+        self.max_sleep = max_sleep
+
+    def backoff(self):
```

Comment:
after a certain incident in my career, I tend to reach for jittered exponential backoff automatically. Linear or non-jittered would probably be fine too, but this has some nice properties and is barely any more complicated.